libmscore/read206.cpp

```diff
@@ -998,6 +999,15 @@ static void readInstrument(Instrument *i, Part* p, XmlReader& e)
             else
                  e.unknown();
             }
+
+      // Read single-note dynamics from template
+      QString templateName = i->trackName().toLower().replace(" ", "-");
+      InstrumentTemplate* tp = searchTemplate(templateName);
+      if (tp)
+            i->setSingleNoteDynamics(tp->singleNoteDynamics);
+      else if (templateName == "")
+            i->setSingleNoteDynamics(false);
```

Comment:
This chunk of code is repeated at least three times here, would it be better to extract it as a separate function?

Reply:
Fixed, `Instrument::setSingleNoteDynamicsFromTemplate();`.